packages/bun-vscode/src/features/tests/bun-test-controller.ts

```diff
@@ -0,0 +1,1539 @@
+            }
+          }
+          if (!errorMsg) errorMsg = `Test "${name}" failed (check "test result" output console for details)`;
+        }
+
+        results.push({
+          name,
+          status,
+          ...(errorMsg !== undefined ? { message: errorMsg } : {}),
+          ...(duration !== undefined && duration > 0 ? { duration } : {}),
+          location: {
+            file: file || (suite.name as string),
+            line,
+            column: 0,
+          },
+          ...(fullParent ? { parent: fullParent } : {}),
+        });
+      }
+    }
+    return results;
+  }
+
+  if (xmlResult && typeof xmlResult === "object" && xmlResult !== null) {
+    const xmlObj = xmlResult as Record<string, unknown>;
+    if (xmlObj.testsuites && typeof xmlObj.testsuites === "object") {
+      const testsuites = xmlObj.testsuites as Record<string, unknown>;
+      if (testsuites.testsuite) {
+        const suites = Array.isArray(testsuites.testsuite) ? testsuites.testsuite : [testsuites.testsuite];
+        for (const suite of suites as Record<string, unknown>[]) {
+          const fileName = suite.name as string;
+          const filePath = path.isAbsolute(fileName) ? fileName : path.resolve(workspacePath, fileName);
+
+          const topLevel = parseXmlSuite(suite);
+          const idx = testResults.findIndex(r => r.name === filePath);
+          if (idx !== -1) testResults.splice(idx, 1);
+
+          testResults.push({
+            name: filePath,
+            tests: topLevel,
+            passed: topLevel.every(t => t.status === "passed"),
+          });
+        }
+      }
+    }
+  }
+
+  return testResults;
+}
+
+function processErrorData({
+  testResult,
+  testItem,
+}: {
+  testResult: BunTestResult;
+  testItem: vscode.TestItem;
+}): vscode.TestMessage | undefined {
+  if (!testResult.message) {
+    return undefined;
+  }
+
+  const messageLinesRaw = testResult.message.split("\n");
+  const stackTraceIndex = messageLinesRaw.findIndex(
+    line =>
+      line.trim().startsWith("at ") &&
+      line.includes("(") &&
+      line.includes(":") &&
+      line.includes(testItem.uri?.fsPath || ""),
+  );
+  let lines: string[];
+  if (stackTraceIndex !== -1) {
+    lines = messageLinesRaw.slice(0, stackTraceIndex + 1).map(line => line);
+  } else {
+    lines = messageLinesRaw.map(line => line);
+  }
+
+  const errorLine = lines[0].trim();
+  const messageLines = lines.slice(1, -1).join("\n");
+  const fileLine = lines.slice(-1)[0].trim();
+
+  const filepath = getFileLocationFromError(fileLine, testItem.uri?.fsPath || "");
+
+  if (!filepath) {
+    return new vscode.TestMessage(testResult.message);
+  }
+
+  const messageLocation = new vscode.Location(
+    vscode.Uri.file(filepath.file),
+    new vscode.Position(filepath.line - 1, filepath.column),
+  );
+
+  const errorType = errorLine.replace(/^(E|e)rror: /, "").trim();
+  switch (errorType) {
+    case "expect(received).toMatchInlineSnapshot(expected)":
```

Comment:
there are a lot more error types, this is very brittle

Reply:
I just wanted to do a few that could make best use of the vscode extension api. They should work otherwise good enough in the fallback one.

If I got a more detailed error that would help, but as I only have the error itself there isn't too much I can do